boto/kms/layer1.py

```diff
@@ -0,0 +1,824 @@
+            disabled. This can be an ARN, an alias, or a globally unique
+            identifier.
+
+        """
+        params = {'KeyId': key_id, }
+        return self.make_request(action='DisableKey',
+                                 body=json.dumps(params))
+
+    def disable_key_rotation(self, key_id):
+        """
+        Disables rotation of the specified key.
+
+        :type key_id: string
+        :param key_id: Unique identifier of the customer master key for which
+            rotation is to be disabled. This can be an ARN, an alias, or a
+            globally unique identifier.
+
+        """
+        params = {'KeyId': key_id, }
+        return self.make_request(action='DisableKeyRotation',
+                                 body=json.dumps(params))
+
+    def enable_key(self, key_id):
+        """
+        Marks a key as enabled, thereby permitting its use. You can
+        have up to 25 enabled keys at one time.
+
+        :type key_id: string
+        :param key_id: Unique identifier of the customer master key to be
+            enabled. This can be an ARN, an alias, or a globally unique
+            identifier.
+
+        """
+        params = {'KeyId': key_id, }
+        return self.make_request(action='EnableKey',
+                                 body=json.dumps(params))
+
+    def enable_key_rotation(self, key_id):
+        """
+        Enables rotation of the specified customer master key.
+
+        :type key_id: string
+        :param key_id: Unique identifier of the customer master key for which
+            rotation is to be enabled. This can be an ARN, an alias, or a
+            globally unique identifier.
+
+        """
+        params = {'KeyId': key_id, }
+        return self.make_request(action='EnableKeyRotation',
+                                 body=json.dumps(params))
+
+    def encrypt(self, key_id, plaintext, encryption_context=None,
+                grant_tokens=None):
+        """
+        Encrypts plaintext into ciphertext by using a customer master
+        key.
+
+        :type key_id: string
+        :param key_id: Unique identifier of the customer master. This can be an
+            ARN, an alias, or the Key ID.
+
+        :type plaintext: blob
+        :param plaintext: Data to be encrypted.
+
+        :type encryption_context: map
+        :param encryption_context: Name:value pair that specifies the
+            encryption context to be used for authenticated encryption. For
+            more information, see `Authenticated Encryption`_.
+
+        :type grant_tokens: list
+        :param grant_tokens: A list of grant tokens that represent grants which
+            can be used to provide long term permissions to perform encryption.
+
+        """
+        if plaintext is not None:
+            if not isinstance(plaintext, six.binary_type):
+                raise TypeError(
+                    "Argument ``plaintext`` is "
+                    "modeled as a blob. Value must be bytes.")
```

Comment:
I suspect users may not understand the `modeled as a blob` bit.  What if we just say something like "value must be of type %s" % six.binary_type?